tests/test-timber.php

```diff
@@ -28,12 +28,92 @@ function testGetPostString(){
 	}
 
 	function testGetPostBySlug(){
-		$this->factory->post->create(array('post_name' => 'kill-bill'));
-		$post = new Timber\Post('kill-bill');
-		$this->assertEquals('kill-bill', $post->post_name);
-		/** Test deprecated method */
-		$post = Timber::get_post('kill-bill');
-		$this->assertEquals('kill-bill', $post->post_name);
+		$this->factory->post->create( [ 'post_name' => 'kill-bill' ] );
+
+		$post = Timber\Timber::get_post_by( 'slug', 'kill-bill');
+
+		$this->assertEquals( 'kill-bill', $post->post_name );
+	}
+
+	function testGetPostBySlugAndPostType(){
+		$post_id_movie = $this->factory->post->create( [
+			'post_name' => 'kill-bill',
+			'post_type' => 'movie',
+		] );
+		$post_id_page  = $this->factory->post->create( [
+			'post_name' => 'kill-bill',
+			'post_type' => 'page',
+		] );
+
+		$post_movie = Timber\Timber::get_post_by( 'slug', 'kill-bill', 'movie' );
+		$post_page  = Timber\Timber::get_post_by( 'slug', 'kill-bill', 'page' );
+
+		$this->assertEquals( $post_id_movie, $post_movie->ID );
+		$this->assertEquals( $post_id_page, $post_page->ID );
+
+		$post_any = Timber\Timber::get_post_by( 'slug', 'kill-bill' );
+		$this->assertEquals( $post_id_movie, $post_any->ID );
```

Comment:
This actually returns the value of `$post_id_page` because it's the newest post (and WP's default sort is `post_date DESC`